ang/civicase/Utils.js

```diff
@@ -553,48 +545,117 @@
-          }
-          else {
-            dropdown.css('left', button.offset().left + "px");
+      link: function (scope, $el, attrs) {
+        function dropDownFixPosition (button, dropdown) {
+          var dropDownTop = -$(window).scrollTop() + button.offset().top + button.outerHeight();
+          dropdown.css('top', dropDownTop + 'px');
+          if (dropdown.hasClass('dropdown-menu-right')) {
+            dropdown.css('left', button.offset().left - dropdown.outerWidth() + button.outerWidth() + 'px');
+          } else {
+            dropdown.css('left', button.offset().left + 'px');
           }
         }
 
-        $timeout(function() {
-          $($el).click(function(){
-            dropDownFixPosition($(this),$(this).next('.dropdown-menu'));
+        $timeout(function () {
+          $($el).click(function () {
+            dropDownFixPosition($(this), $(this).next('.dropdown-menu'));
           });
 
-          document.addEventListener('scroll', function(e){
-            dropDownFixPosition($($el),$($el).next('.dropdown-menu'));
+          document.addEventListener('scroll', function (e) {
+            dropDownFixPosition($($el), $($el).next('.dropdown-menu'));
           }, true);
         });
       }
     };
   });
 
-  angular.module('civicase').directive('calculateScrollWidth', function() {
+  /**
+   * Directive for the sticky table header functionality on case list page
+   */
+  module.directive('stickyTableHeader', function () {
+    return {
+      restrict: 'A',
+      link: function (scope, $el, attrs) {
+        var $table = $('.civicase__list');
+
+        // Watch if loading completes
+        scope.$watch(function (scope) {
+          return !scope.isLoading;
+        },
+        function (completed) {
+          if (completed) { // loading complete
+            var bodyPadding = parseInt($('body').css('padding-top'), 10); // to see the space for fixed menus
+            var topPos = $el.offset().top - bodyPadding;
+            // Assign min-width values to th to have solid grid
+            $('th', $el).each(function () {
+              $(this).css('min-width', $(this).outerWidth() + 'px');
+            });
+
+            // Define when to make the element sticky (affixed)
+            $($el).affix({
+              offset: {
+                top: topPos
+              }
+            })
+            // After element is affixed set scrolling pos (to avoid glitch) and top position
+              .on('affixed.bs.affix', function () {
+                $el.scrollLeft($table.scrollLeft());
+                $el.css('top', bodyPadding + 'px');
+              });
+            // Attach scroll function
+            $table.scroll(function () {
+              $el.scrollLeft($(this).scrollLeft());
+            });
+          }
+        });
+      }
+    };
+  });
+
+  /**
+   * Directive for the sticky pager footer functionality on case list page
+   */
+  module.directive('stickyFooterPager', function () {
```

Comment:
Name is generic but you are using case classes inside. Lets discuss

Reply:
Class used are `civicase__pager--fixed` which is not dependent upon any sub component of civicase. Please let me know if we still need some discussions.